crates/kornia-imgproc/src/threshold.rs

```diff
@@ -374,11 +374,72 @@ where
 
 // TODO: outsu, triangle
 
+/// Apply an otsu threshold to an image
+pub fn threshold_otsu<A1: ImageAllocator, A2: ImageAllocator>(
+    src: &Image<u8, 1, A1>,
+    dst: &mut Image<u8, 1, A2>,
+    threshold_out: &mut u8,
+) -> Result<(), ImageError> {
+    if src.size() != dst.size() {
+        return Err(ImageError::InvalidImageSize(
+            src.cols(),
+            src.rows(),
+            dst.cols(),
+            dst.rows(),
+        ));
+    }
+    let mut hist: Vec<usize> = vec![0_usize; 256];
+    let _ = compute_histogram(src, &mut hist, 256).unwrap();
+
+    let n = (src.rows() * src.cols()) as f64;
+
+    let mut sum = 0f64;
+    for i in 0..256 {
```

Comment:
```suggestion
    for i in 0..HIST_SIZE {
```
no magic numbers, use consts. same for the rest